scripts/classify-orgs-and-cves/classify-cves.py

```diff
@@ -0,0 +1,323 @@
+
+    Parameters
+    ----------
+    `classifier`: zero-shot classifier model.
+    `vulnSummary`: CVE summary to be classified, in English.
+
+    Returns
+    -------
+    `result`: dictionary with the classification result.
+    """
+
+    # Summary is already in EN, we don't need to translate
+
+    # Get sequence to classify and add some context
+    sequence: str = f"{vulnSummary}"
+    hypothesis: str = "This summary is from a {} vulnerability."
+
+    result: dict = classifier(
+        sequences=sequence,
+        candidate_labels=VULN_LABELS,
+        hypothesis_template=hypothesis,
+        multi_label=False,
+    )
+
+    return result
+
+
+def classifyCveConfiguration(classifier: Pipeline, vulnSummary: str) -> dict[str, Any]:
+    """
+    Classifies a CVE into a configuration category.
+
+    That is, if the vulnerability is active in the default configuration of a software, or not.
+
+    The classification is not binary and produces a vector of probabilities.
+
+    Parameters
+    ----------
+    `classifier`: zero-shot classifier model.
+    `vulnSummary`: CVE summary to be classified, in English.
+
+    Returns
+    -------
+    `result`: dictionary with the classification result.
+    """
+
+    # Summary is already in EN, we don't need to translate
+
+    # Get sequence to classify and add some context
+    sequence: str = f"{vulnSummary}"
+    hypothesis: str = "This vulnerability is {}."
+
+    result: dict = classifier(
+        sequences=sequence,
+        candidate_labels=CONFIG_LABELS,
+        hypothesis_template=hypothesis,
+        multi_label=False,
+    )
+
+    return result
+
+
+def processCvesFromMitre(
+    filepath: str, classifier: Pipeline
+) -> dict[str, tuple[str, dict[str, float]]]:
+    """
+    Process all CVEs from a Mitre zip file and classify them using a zero-shot classifier.
+
+    The CVEs are classified into vulnerability categories and further classified into configuration categories.
+
+    Parameters
+    ----------
+    `filepath`: path to the zip file containing the CVEs.
+    `classifier`: zero-shot classifier model.
+
+    Returns
+    -------
+    `result`: dictionary with the classification result for each CVE.
+    """
+
+    seenCves: dict[str, tuple[str, dict[str, float]]] = dict()
+
+    with zipfile.ZipFile(filepath, "r") as file:
+        # The zip file contains a folder with all the CVEs as JSON files
+        cveFilename: list[str] = file.namelist()
+
+        for idx, cveFile in enumerate(cveFilename):
+            if idx % 1000 == 0:
+                logging.info(f"Progress {idx + 1} / {len(cveFilename)}")
+
+            # There are some files that are not JSON, they do not contain CVEs
+            if not cveFile.endswith(".json"):
+                continue
+
+            with file.open(cveFile) as f:
+                data: dict[str, Any] = json.loads(f.read())
+
+                try:
+                    state: str = data["cveMetadata"]["state"]
+                except:
```

Comment:
```suggestion
                except KeyError:
```

Or is it something else?

Reply:
That is it, I also updated the two getters a few lines below.

Reply:
EDIT: not quite, if state is not found, then the other fields might have different exceptions. I have left it as except only.

Reply:
Maybe `except KeyError, IndexError`? There doesn't seem to be anything else going on here... unless we have overloaded `__getitem__` throwing unexpected exceptions around.

Reply:
Second exception is actually a TypeError, so `except (KeyError, TypeError)` works like a charm now.